src/components/orator/js/Orator.js

```diff
@@ -0,0 +1,963 @@
+            target: "{that selectionReader}.options",
+            removeSource: true,
+            namespace: "selectionReaderOpts"
+        }]
+    });
+
+    fluid.orator.handlePlayToggle = function (that, state) {
+        if (state) {
+            that.play();
+        } else {
+            that.pause();
+        }
+    };
+
+    /**********************************************
+     * fluid.orator.controller
+     *
+     * Provides a UI Widget to control the Orator
+     **********************************************/
+
+    fluid.defaults("fluid.orator.controller", {
+        gradeNames: ["fluid.containerRenderingView"],
+        selectors: {
+            playToggle: ".flc-orator-controller-playToggle"
+        },
+        styles: {
+            play: "fl-orator-controller-play"
+        },
+        strings: {
+            play: "play",
+            pause: "pause"
+        },
+        model: {
+            playing: false,
+            enabled: true
+        },
+        markup: {
+            container: "<div class=\"flc-orator-controller fl-orator-controller\">" +
+                "<div class=\"fl-icon-orator\" aria-hidden=\"true\"></div>" +
+                "<button class=\"flc-orator-controller-playToggle\">" +
+                    "<span class=\"fl-orator-controller-playToggle fl-icon-orator-playToggle\" aria-hidden=\"true\"></span>" +
+                "</button></div>"
+        },
+        invokers: {
+            play: {
+                changePath: "playing",
+                value: true,
+                source: "play"
+            },
+            pause: {
+                changePath: "playing",
+                value: false,
+                source: "pause"
+            },
+            toggle: {
+                funcName: "fluid.orator.controller.toggleState",
+                // Not providing an option for the explicit state value because
+                // when called through the jQuery click event the event object
+                // is passed in.
+                args: ["{that}", "playing"]
+            },
+            // overriding the addToParent to "prepend" the container to the parentContainer
+            addToParent: {
+                args: ["{that}.options.parentContainer", "{arguments}.0", "prepend"]
+            }
+        },
+        listeners: {
+            "onCreate.bindClick": {
+                "this": "{that}.dom.playToggle",
+                method: "click",
+                args: ["{that}.toggle"]
+            }
+        },
+        modelListeners: {
+            "playing": {
+                listener: "fluid.orator.controller.setToggleView",
+                args: ["{that}", "{change}.value"]
+            },
+            "enabled": {
+                "this": "{that}.container",
+                method: "toggle",
+                args: ["{change}.value"],
+                namespace: "toggleView"
+            }
+        }
+    });
+
+    fluid.orator.controller.toggleState = function (that, path, state) {
```

Comment:
Docs for this function given its requirements seem a bit loose - might be helpful to mention it is designed to be invoked directly as a click handler. What is the reason for the odd polymorphism based on accepting the state either from the direct argument or from the model?